src/inference/src/ie_core.cpp

```diff
@@ -881,40 +881,7 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
         OV_ITT_SCOPED_TASK(ov::itt::domains::IE, "Core::QueryNetwork");
         auto parsed = parseDeviceNameIntoConfig(deviceName, config);
         auto res = GetCPPPluginByName(parsed._deviceName).query_model(network, parsed._config);
-        if (!network.getFunction() || res.supportedLayersMap.empty())
```

Comment:
Why do we need to move this code to the plugins?
Do you sure that we don't need this code anymore?

Reply:
Hi, ilyachur, according to the feedback from the HPU plugin, the check here will lead to some sub graph segmentation errors.
To ensure that it is feasible to remove the code, we add the corresponding check to the querynetwork of each plugin.

Reply:
Do we know cases when this code leads any issues for HPU? Maybe first of all we need to take a look to these cases.

Reply:
@WeldonWangwang I agree with @ilyachur we need to look at error at first to understand what causing the problem.

Reply:
@ilyachur @nkogteva , I will confirm again what is the problem causing the HPU issues.

Reply:
> @ilyachur @nkogteva , I will confirm again what is the problem causing the HPU issues.

Could you please provide any details?

Reply:
@ilyachur @nkogteva Running ConstantFolding() in ie_core is common w/a for QueryNetwork() of HW plugin when QueryNetworkResult of HW plugin miss supported constant ops. This w/a expect constant ops always supported by each HW plugin.
Then this w/a will cause issue in below case:

1.  QueryNetwork() filters supported ops not only based on the ops level, but also on the sub-graph level. When one ops A in one sub-graph cannot be supported by HW, constant ops B in this sub-graph would not be support by this HW. Then QueryNetworkResult would not include this constant ops B.
2. Running ConstantFolding() in ie_core is based on ops level only. Since constant ops B can be supported by HW, ie_core will add constant ops B into QueryNetworkResult. 
3. HETERO will separate network based on updated QueryNetworkResult which is not HW expectation. May cause too many sub-graphs and data transfer.

The point is HW pluing should know how to separate network for this HW and return right QueryNetworkResult. HETERO or ie core should not modify the result. If there is any w/a needed, the w/a should be in each plugin. This PR just move w/a from ie core to each exist plugin for this point.

Reply:
СС @slyalin @jane-intel

Reply:
@ilyachur I found the ticket which was originally fixed by this code. I removed this code from core part without adding into the plugin and I don't see any problems when use old reproducer. So probably this WA is not needed anymore and we don't need to copy past this code to plugins. It would be nice to gather statictis on wider scope of topologies with and without workaround.

Reply:
@nkogteva Based on limited test with OMZ models, looks like WA can help some networks on MyraidX. Since we have not tested more networks or fully understand the case WA covered, we move WA to each existing plugin first to avoid any regression issue. After that, we can have follow up PR to remove WA in any plugin after we fully understand the impact.

Reply:
> @nkogteva Based on limited test with OMZ models, looks like WA can help some networks on MyraidX. Since we have not tested more networks or fully understand the case WA covered, we move WA to each existing plugin first to avoid any regression issue. After that, we can have follow up PR to remove WA in any plugin after we fully understand the impact.

If there are problems in MyriadX, ticket should be created to appropriate plugin. Otherwise we just copy pastes WA over all plugins, where it is not necessary probably.